### PR TITLE
feat(components): add documentation to all exported types

### LINKED ISSUE #103: Add documentation to all exported types
MAD components is missing a lot of documentation. We start this off by writing docstrings to all props that we expose to the end-user. Should perhaps also keep this in mind when making new components going forward..

### PR DESCRIPTION
**Pull request by HaakonSvane**
- Write docstrings for all exported types and methods
- Remove `test` prop on `LinearProgress` indicator component

closes #103

### CODE REVIEW

**Comment by HaakonSvane** [comment]
Don't be alarmed by all the files changed 🥇

**Review by tormodAase — CHANGES_REQUESTED** [review]
Good shit! The typos are probably nice to get rid of, and I have some improvement proposals regarding typography bold/italic props.

File: packages/components/src/components/Spacer/types.ts [hunk 1]
```diff
@@ -1,3 +1,6 @@
 export type SpacerProps = {
+    /**
+     * The amount of spacing to apply to the spacer.
+     */
     amount?: "small" | "medium" | "large";
-};
\ No newline at end of file
+};
```
[the remark this answers is not in the document]
Comment (tormodAase) [reply packages/components/src/components/Spacer/types.ts:6]: Maybe add a `@see https://eds.equinor.com/0b0c666ab/p/721e1f-spacing/b/0899da`.
Also wouldn't hurt to tell the user what it defaults to

File: packages/components/src/components/Button/ButtonGroup.tsx [hunk 2]
```diff
@@ -3,11 +3,20 @@ import { StyleSheet, View, ViewProps } from "react-native";
...
 };
 
 export type ButtonGroupContextType = {
+    /**
+     * Indicates that the contexed item is the first item in the button group. 
+     */
     isFirstItem: boolean;
+    /**
+     * Indicates that the contexed item is the last item in the button group.
+     */
```
[the remark this answers is not in the document]
Comment (HaakonSvane) [reply packages/components/src/components/Button/ButtonGroup.tsx]: fixed in b7711ae

File: packages/components/src/components/Cell/index.ts [hunk 3]
```diff
@@ -4,18 +4,24 @@ import { Cell as _Cell, CellProps } from "./Cell";
 import { CellSwipeItemProps } from "./types";
 
 type CellFamily = typeof _Cell & {
+    /**
+     * A container for grouping cells togehter. The child cells visually come together inside a group.
```
[the remark this answers is not in the document]
Comment (HaakonSvane) [reply packages/components/src/components/Cell/index.ts]: fixed in f62e1ec

File: packages/components/src/components/Input/Input.tsx [hunk 4]
```diff
@@ -5,14 +5,43 @@ import { Label, useStyles } from "../..";
 import React from "react";
 
 export type InputProps = {
+    /**
+     * A small label text to add to the input field.
+     */
     label?: string;
+    /**
+     * Secondary small label text to add tot he input field.
```
[the remark this answers is not in the document]
Comment (HaakonSvane) [reply packages/components/src/components/Input/Input.tsx]: fixed in 7c646f1

File: packages/components/src/components/Portal/PortalContext.tsx [hunk 5]
```diff
@@ -1,15 +1,37 @@
 import React, { PropsWithChildren, ReactNode, createContext, useState } from "react";
 
 export type PortalContextType = {
+    /**
+     * Register a new host in the context.
+     * @param name The name of the portal to register.
+     */
     registerHost: (name: string) => void;
+    /**
+     * Deletes the portal from the registery.
```
[the remark this answers is not in the document]
Comment (HaakonSvane) [reply packages/components/src/components/Portal/PortalContext.tsx]: fixed in d76721b

File: packages/components/src/components/Spacer/types.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (HaakonSvane) [reply packages/components/src/components/Spacer/types.ts]: fixed in 25cd7d5

**Review by tormodAase — APPROVED** [review]

**Pull request merged**